crates/miden-objects/src/account/storage/header.rs

```diff
@@ -90,6 +92,11 @@ impl AccountStorageHeader {
         })
     }
 
+    /// Computes the account storage header commitment.
+    pub fn compute_commitment(&self) -> Digest {
+        Hasher::hash_elements(&self.as_elements())
```

Comment:
There is some duplication of logic between this function and `build_slots_commitment()` in the `super`. At the very lest, we should add comments to make sure these two implementations stay in sync.

Reply:
I added a comment that explains that they should be in sync. I think there might be room for some more refactors there as well but decided not to include them at this point.